src/modules/blockchain/implementation/web3-service.js

```diff
@@ -269,6 +272,31 @@ class Web3Service {
         }
     }
 
+    cacheParameter(contractName, parameterName, parameterValue) {
+        const found =
+            CACHED_FUNCTIONS[contractName] &&
+            CACHED_FUNCTIONS[contractName].find((item) => item.name === parameterName);
+        if (found) {
+            const { type } = found;
+            switch (type) {
```

Comment:
Following the suggestion to remove types from constants, I would make it a separate function somewhere in the `services` or even `utils` for casting based on types of the event arguments in the ABI and cast it directly in the event handler instead of doing it here